lib/puppet/provider/registry_value/registry.rb

```diff
@@ -174,6 +174,57 @@ def write_value
     end
   end
 
+
+  def data_to_bytes(type, data, &blk)
+    bytes = []
+
+    case type
+      when Win32::Registry::REG_SZ, Win32::Registry::REG_EXPAND_SZ
+        bytes = wide_string(data).bytes
+      when Win32::Registry::REG_MULTI_SZ
+        # each wide string is already NULL terminated
+        bytes = data.map { |s| wide_string(s).bytes }.flat_map { |a| a }
+        # requires an additional NULL terminator to terminate properly
+        bytes << 0 << 0
+      when Win32::Registry::REG_BINARY
+        # TODO: not sure what to do here - old code did data.to_s
+        bytes = wide_string(data).bytes
```

Comment:
I don't think we want to call `wide_string` here as that will try to interpret the caller supplied values as wide characters. Instead, I'd think we want to do `bytes = data.bytes` essentially treating the string as a binary data.